heat/core/linalg/tests/test_lanczos.py

```diff
@@ -0,0 +1,87 @@
+import torch
+import os
+import unittest
+import heat as ht
+import numpy as np
+
+if os.environ.get("DEVICE") == "gpu" and torch.cuda.is_available():
+    ht.use_device("gpu")
+    torch.cuda.set_device(torch.device(ht.get_device().torch_device))
+else:
+    ht.use_device("cpu")
+device = ht.get_device().torch_device
+ht_device = None
+if os.environ.get("DEVICE") == "lgpu" and torch.cuda.is_available():
+    device = ht.gpu.torch_device
+    ht_device = ht.gpu
+    torch.cuda.set_device(device)
+
+
+class TestLinalgLanczos(unittest.TestCase):
```

Comment:
Shouldnt this rather be called optimize?

Reply:
What do you mean?

Reply:
Well the Unittest class is called TestLinalgLanczos. I would go with TestLanczos and remove the cg_test cases

Reply:
Ah, I see. Ok, that I can do

Reply:
I went with solver now. That cover both Lanczos and CG and leaves room for other algorihms (arnoldi, powermethod, etc)